src/loaders/index.ts

```diff
@@ -0,0 +1,57 @@
+import { LoadSpec, loadSpecToString, IVolumeLoader, VolumeDims } from "./IVolumeLoader";
+
+import { OMEZarrLoader } from "./OmeZarrLoader";
+import { JsonImageInfoLoader } from "./JsonImageInfoLoader";
+import { TiffLoader } from "./TiffLoader";
+import VolumeCache from "../VolumeCache";
+
+const enum VolumeFileFormat {
+  ZARR = "zarr",
+  JSON = "json",
+  TIFF = "tiff",
+}
+
+type CreateZarrLoaderOptions = { path: string; cache?: VolumeCache; scene?: number };
+type CreateJsonLoaderOptions = { path: string | string[]; cache?: VolumeCache };
+type CreateTiffLoaderOptions = { path: string };
+
+type CreateLoaderOptionsMap = {
+  [T in VolumeFileFormat]: {
+    [VolumeFileFormat.ZARR]: CreateZarrLoaderOptions;
+    [VolumeFileFormat.JSON]: CreateJsonLoaderOptions;
+    [VolumeFileFormat.TIFF]: CreateTiffLoaderOptions;
+  }[T];
+};
+
+type CreateLoaderOptions = CreateZarrLoaderOptions | CreateJsonLoaderOptions | CreateTiffLoaderOptions;
+
+async function createVolumeLoader<T extends VolumeFileFormat>(
+  fileType: T,
+  options?: CreateLoaderOptionsMap[T]
+): Promise<IVolumeLoader> {
+  switch (fileType) {
+    case VolumeFileFormat.ZARR:
+      const zOptions = options as CreateZarrLoaderOptions;
```

Comment:
"unexpected lexical declaration in case block" might need curly brackets around the block